File: frontend/src/app/drugcomponent/strength.ts
import {Strengthunit} from "./strengthunit";

export class Strength {
  private _strengthId;
  private _strengthUnit: Strengthunit;
  private _strengthAmount;

  get strengthId() {
    return this._strengthId;
  }

  set strengthId(value) {
    this._strengthId = value;
  }

  get strengthUnit(): Strengthunit {
    return this._strengthUnit;
  }

  set strengthUnit(value: Strengthunit) {
    this._strengthUnit = value;
  }

  get strengthAmount() {
    return this._strengthAmount;
  }

  set strengthAmount(value) {
    this._strengthAmount = value;
  }
}

File: src/main/java/com/opdapp/model/Strength.java
package com.opdapp.model;

import javax.persistence.*;

@Entity
public class Strength {

    @Id
    @GeneratedValue
    private int strengthId;

    @ManyToOne
    @JoinColumn(name = "unitId")
    private StrengthUnit strengthUnit;

    private double strengthAmount;

    public StrengthUnit getStrengthUnit() {
        return strengthUnit;
    }

    public void setStrengthUnit(StrengthUnit strengthUnit) {
        this.strengthUnit = strengthUnit;
    }

    public int getStrengthId() {
        return strengthId;
    }

    public void setStrengthId(int strengthId) {
        this.strengthId = strengthId;
    }

    public double getStrengthAmount() {
        return strengthAmount;
    }

    public void setStrengthAmount(double strengthAmount) {
        this.strengthAmount = strengthAmount;
    }
}
